src/main/java/bg/libapp/libraryapp/model/mappers/BookMapper.java

```diff
@@ -0,0 +1,88 @@
+package bg.libapp.libraryapp.model.mappers;
+
+import bg.libapp.libraryapp.model.dto.author.AuthorBookViewDTO;
+import bg.libapp.libraryapp.model.dto.book.BookAddDTO;
+import bg.libapp.libraryapp.model.dto.book.BookAuthorViewDTO;
+import bg.libapp.libraryapp.model.dto.book.BookViewDTO;
+import bg.libapp.libraryapp.model.entity.Author;
+import bg.libapp.libraryapp.model.entity.Book;
+import bg.libapp.libraryapp.repository.AuthorRepository;
+import bg.libapp.libraryapp.repository.GenreRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.stream.Collectors;
+
+@Component
+public class BookMapper {
+    private final AuthorRepository authorRepository;
+    private final GenreRepository genreRepository;
+    private final AuthorMapper authorMapper;
+    private final GenreMapper genreMapper;
+
+    @Autowired
+    public BookMapper(AuthorRepository authorRepository, GenreRepository genreRepository, AuthorMapper authorMapper, GenreMapper genreMapper) {
+        this.authorRepository = authorRepository;
+        this.genreRepository = genreRepository;
+        this.authorMapper = authorMapper;
+        this.genreMapper = genreMapper;
+    }
+
+    public BookAuthorViewDTO mapBookAuthorViewDTOFromBook(Book book) {
+        return new BookAuthorViewDTO()
+                .setIsbn(book.getIsbn())
+                .setYear(book.getYear())
+                .setTitle(book.getTitle())
+                .setPublisher(book.getPublisher())
+                .setGenres(book.getGenres()
+                        .stream()
+                        .map(genreMapper::mapGenreBookViewDTOFromGenre)
+                        .collect(Collectors.toSet()));
+    }
+
+    public Book mapBookFromBookAddDTO(BookAddDTO bookAddDTO) {
```

Comment:
Mappers shouldnt be responsible for creating new entities. Would be best to move the find/create logic of author in author service.